### ISSUE TITLE
Displays/outputs not rescanned after resuming from hibernation

### ISSUE THREAD

**Issue by jamadazi** [swaywm/sway#6388]
- **Sway Version:** 1.6.1
- **Driver:** AMDGPU (Vega 64)
- **Description:**

I often hibernate my desktop computer to move it between desks. I hibernate while it is connected to my monitors on one desk, then unplug all cables, replug everything on the other desk, and resume.

When the system resumes, sway still thinks that the old monitors are connected. The desktop is rendered at the old resolution (and any rotation/transform settings). Running `swaymsg -t get_outputs` lists the old monitors (their names/IDs and video modes), not the new ones.

My current workaround for the problem is to wait until the system has resumed, physically unplug the monitor cables from the computer, and replug them. Sway then detects the newly-hotplugged displays correctly.

**Comment by progandy** [comment]
Is the edid reported by the kernel correct while sway shows the problem?, e.g.

    # https://git.linuxtv.org/edid-decode.git/ | also in arch linux aur
    edid-decode /sys/class/drm/card0-DP-1/edid

Are the monitors detected correctly if you restart sway while the monitors are in the broken state? 
If not, then it may be a kernel bug I think (the old state could be loaded from the hibernation image without hotplug detection afterwards) Possibly try a suspend/resume cycle after resuming from hibernation as well.

You could also try to retrigger udev for the gpu with something like one of these commands, but I think that can only help when the edid reported by the kernel is correct.

```
(sudo) udevadm trigger --action change /sys/class/drm/card0
(sudo) udevadm trigger --action change /sys/class/drm/card0-DP-1
```

**Comment by jamadazi** [comment]
Yes, the edid in the kernel is confirmed correct. The GPU/driver/kernel is behaving correctly.

It seems most plausible that your suspicion about wlroots is the correct one. I'll file a bug there.

**Comment by Emantor** [comment]
I took this as an opportunity to pick swaywm/wlroots#1810 back up. Please test if you can afford the time.

**Comment by jpetazzo** [comment]
It looks like I ran into the same problem: I have two identical monitors arranged side by side, and randomly, when resuming from suspend, they get swapped.

`swaymsg -pt get_outputs` shows that sway¹ got them reversed - i.e., looking at the serial numbers of the screens, it shows e.g. `Output DP-2 'Lenovo Group Limited P32p-30 V30AV5HA' (focused)` while the focused window is, in fact, on the other monitor (confirmed by taking a pic showing both the output of that command, and the serial number of the monitor thanks to the monitor's OSD, because I thought I was hallucinating😅)

The monitors are connected over Thunderbolt, and I suspect that in some cases, when resuming from suspend, `DP-1` and `DP-2` get swapped, and sway¹ hasn't rescanned the outputs to refresh the monitors connected to them.

Reloading sway doesn't help. Restarting it entirely does fix the problem, though.

Since the problem happens in a random way, I unfortunately don't have a way to reliably reproduce it (and efficiently test a PR) but I'm leaving this here to both confirm that the bug still seems here, and to be notified if there is any change.

FWIW it looks like the wlroots PR has been moved to https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/1810 and I'll see if I can subscribe to that too :)

¹After reading the current issue, I understand the problem likely comes from wlroots rather than sway!

Thanks!
